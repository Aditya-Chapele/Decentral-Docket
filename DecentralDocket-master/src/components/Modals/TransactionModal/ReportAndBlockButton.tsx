/*
Copyright 2019-2022 (c) Dappros Ltd, registered in England & Wales, registration number 11455432. All rights reserved.
You may not use this file except in compliance with the License.
You may obtain a copy of the License at https://github.com/dappros/ethora/blob/main/LICENSE.
Note: linked open-source libraries and components may be subject to their own licenses.
*/

import * as React from "react"
import { Text, View, StyleSheet, TouchableOpacity } from "react-native"
import MaterialIcons from "react-native-vector-icons/MaterialIcons"
import { textStyles } from "../../../../docs/config"
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen"

interface ReportAndBlockButtonProps {
  onPress: any
  text: string
  style?: any
}

const ReportAndBlockButton = (props: ReportAndBlockButtonProps) => {
  const iconName = "block"
  return (
    <TouchableOpacity onPress={props.onPress} accessibilityLabel={props.text}>
      <View style={[styles.reportAndBlockContainer, props.style]}>
        <View style={styles.blockIcon}>
          <MaterialIcons name={iconName} size={15} color="#fff" />
        </View>
        <Text style={styles.reportAndBlockText}>{props.text}</Text>
      </View>
    </TouchableOpacity>
  )
}

export default ReportAndBlockButton

const styles = StyleSheet.create({
  reportAndBlockContainer: {
    width: wp("50%"),
    height: hp("5%"),
    borderRadius: hp("1%"),
    margin: 5,
    justifyContent: "center",
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#B22222",
    paddingHorizontal: 10,
  },
  reportAndBlockText: {
    fontFamily: textStyles.semiBoldFont,
    color: "#fff",
  },
  blockIcon: {
    marginRight: 5,
    marginBottom: 3,
  },
})
