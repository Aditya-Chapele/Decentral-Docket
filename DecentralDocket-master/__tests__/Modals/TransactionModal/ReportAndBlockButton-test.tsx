import React from "react"
import renderer from "react-test-renderer"
import ReportAndBlockButton from "../../../src/components/Modals/TransactionModal/ReportAndBlockButton"

test("renders correctly", () => {
  const tree = renderer
    .create(
      <ReportAndBlockButton onPress={() => console.log("press")} text="text" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
