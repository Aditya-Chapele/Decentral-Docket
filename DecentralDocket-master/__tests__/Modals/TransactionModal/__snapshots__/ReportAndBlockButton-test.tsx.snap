// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityLabel="text"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#B22222",
          "borderRadius": 13.5,
          "flexDirection": "row",
          "height": 66.5,
          "justifyContent": "center",
          "margin": 5,
          "paddingHorizontal": 10,
          "width": 375,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "marginBottom": 3,
          "marginRight": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#fff",
              "fontSize": 15,
            },
            undefined,
            {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î…‹
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#fff",
          "fontFamily": "Poppins-SemiBold",
        }
      }
    >
      text
    </Text>
  </View>
</View>
`;
